azure/scope/machine.go

```diff
@@ -278,14 +278,12 @@ func (m *MachineScope) AvailabilityZone() string {
 
 // Name returns the AzureMachine name.
 func (m *MachineScope) Name() string {
+	if id := m.GetVMID(); id != "" {
```

Comment:
this is what allows this to be backward compatible

Reply:
nice! could we add a test for this?

Reply:
added